# Seattle Kids in Urban Centers

## Problem Statement:

## Background
Welcome to Seattle, the big city with a small town feeling.  Though Seattle is by all acounts a major metropolitan area, the small neighborhood focus of the city lends to a small town feeling.  These higher-density urban areas within the city are not an accident but rather an approach to focusing development in specific areas of the city, along corridors with transportation, to avoid rezoning large fractions of residential land. (add citation) Today, most of the residential land in the city remains zoned exclusively for single-family homes.  (add citation)

In July 2019, the city enacted legislation allowing the construction of accessory dwelling units (ADUs) on single-family zoned property. (add citation) This change was enacted to help with a housing affordability crisis the city was facing.  The housing crisis continues despite the increase in ADUs and the city is currently examining zoning restrictions in all areas of the city on a neighborhood-by-neighborhood basis.  This rezoning has focused on the urban centers and villages 

The city of Seattle has seen population growth over the last twenty years.  While most residential land in the city has been zoned for single-family housing the explosive growth of the last two decades has resulted in significant growth in the more urban areas of the city.  The city designates these areas as 32 individual 'Urban Centers and Villages.'  The centers are subject to different zoning than the surrounding neighborhood and are considered targets for dense development.  

## Data Collection

Population data was retrieved from the city of Seattle website on December 5.  The population data is US census data aggregated into city-designated areas.  

Population of children in each given Urban Center was gleaned from city reports retrieved on December 5.  These reports are based on early results from the 2020 census, aggregated by city-designated urban centers.  


In [1]:
#imports
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt

import dash
from dash import dcc
from dash import html
import plotly.express as px

In [2]:
# get population data
ucvpop = pd.read_csv("./data/citydata/Urban_Centers_and_Villages_with_PL_94-171_Redistricting__Data_for_1990-2020.csv")
ucvpop.set_index('OBJECTID', inplace = True)
ucvpop.head()

NEIGH_NO               NEIGH_NAME UV_TYPE  \
OBJECTID                                              
1              106         Upper Queen Anne     RUV   
2              302                  Fremont     HUV   
3              205    West Seattle Junction     HUV   
4              111            Columbia City     RUV   
5              414  First Hill/Capitol Hill      UC   

                          TYPE_NAME  AREA_ACRES  F1990_PL_data_TOT_POP  \
OBJECTID                                                                 
1         Residential Urban Village   52.658720                   1921   
2                 Hub Urban Village  214.511083                   3153   
3                 Hub Urban Village  270.417026                   3627   
4         Residential Urban Village  334.543340                   4281   
5                      Urban Center  916.221801                  28940   

          F1990_PL_data_WHITE_NOT  F1990_PL_data_BLACK_NOT  \
OBJECTID                                                     
1                            1715                       56   
2                            2697                       90   
3                            3169                      145   
4                             898                     1827   
5                           21504                     3037   

          F1990_PL_data_AMI_NOT  F1990_PL_data_ASIAN_NOT  ...  \
OBJECTID                                                  ...   
1                            11                       71  ...   
2                            64                      189  ...   
3                            33                      150  ...   
4                           111                     1269  ...   
5                           534                     2403  ...   

          F1990_PL_data_GQ_I  F1990_PL_data_GQ_N_I  F2000_PL_data_GQ_I  \
OBJECTID                                                                 
1                          0                    37                   0   
2                          0                    47                   0   
3                        401                    14                 358   
4                          0                     5                  98   
5                        660                  1370                 782   

          F2000_PL_data_GQ_N_I  F2010_PL_data_GQ_I  F2010_PL_data_GQ_N_I  \
OBJECTID                                                                   
1                            0                   0                     4   
2                           63                   0                    90   
3                           13                 272                     0   
4                            7                  95                    20   
5                         2278                 332                  2329   

          F2020_PL_data_GQ_I  F2020_PL_data_GQ_N_I   Shape__Area  \
OBJECTID                                                           
1                          0                     0  2.293805e+06   
2                          0                    66  9.344065e+06   
3                        235                    12  1.177932e+07   
4                        102                    80  1.457265e+07   
5                        846                  3135  3.991046e+07   

          Shape__Length  
OBJECTID                 
1           11354.22008  
2           16236.22258  
3           20674.93070  
4           25210.67846  
5           31324.33554  

[5 rows x 110 columns]

In [3]:
ucvpop.shape

(32, 110)

In [4]:
ucvpop.isna().sum().sum()

0

The data has no nulls.  We have 110 columns for each of the 32 Urban Centers

## First look at overall population changes
Adding a `change` feature that represents the change in population from 2010 to 2020 and a `perc_change` that represents the percentage change in population from 2010 to 2020

In [5]:
ucvpop['change'] = ucvpop['F2020_PL_data_TOT_POP'] - ucvpop['F2010_PL_data_TOT_POP']

In [6]:
ucvpop['perc_change'] = (ucvpop['F2020_PL_data_TOT_POP'] - ucvpop['F2010_PL_data_TOT_POP'])/ucvpop['F2010_PL_data_TOT_POP']

In [7]:
ucvpop[['NEIGH_NAME', 'AREA_ACRES','F2010_PL_data_TOT_POP','F2020_PL_data_TOT_POP', 'change', 'perc_change']].sort_values('perc_change', ascending=False)

NEIGH_NAME   AREA_ACRES  F2010_PL_data_TOT_POP  \
OBJECTID                                                                  
7                  South Lake Union   339.886927                   3774   
3             West Seattle Junction   270.417026                   5043   
6                         Roosevelt   170.211776                   2616   
12                 Greater Duwamish  4953.281793                   1064   
4                     Columbia City   334.543340                   4609   
20                           Uptown   333.000698                   7300   
19                         Downtown   953.281747                  26844   
30                          Ballard   495.476305                  11440   
2                           Fremont   214.511083                   3960   
13                   Madison-Miller   145.323420                   4066   
5           First Hill/Capitol Hill   916.221801                  35892   
14                       Green Lake   107.444638                   2904   
23              Bitter Lake Village   363.779566                   4519   
8               University District   751.963878                  22875   
11                        Lake City   142.267043                   3899   
22             23rd & Union-Jackson   625.266413                  10460   
25                          Admiral    98.248351                   1528   
24                         Mt Baker   491.018058                   5325   
1                  Upper Queen Anne    52.658720                   2143   
32                          Othello   498.963241                   9330   
15                        Northgate   412.291123                   6620   
28                         Eastlake   199.255956                   5084   
29                      Wallingford   258.005613                   5425   
18          Greenwood-Phinney Ridge    94.157645                   2927   
21                    Rainier Beach   346.143874                   4094   
9                        Crown Hill   270.988754                   4191   
17        Ballard-Interbay-Northend   931.972864                   1598   
27                  Morgan Junction   113.309377                   2046   
26            Aurora-Licton Springs   326.958599                   6179   
16                North Beacon Hill   266.876291                   5217   
10           Westwood-Highland Park   275.062196                   4606   
31                       South Park   263.798620                   3448   

          F2020_PL_data_TOT_POP  change  perc_change  
OBJECTID                                              
7                         12905    9131     2.419449  
3                          9160    4117     0.816379  
6                          4678    2062     0.788226  
12                         1836     772     0.725564  
4                          7661    3052     0.662183  
20                        11840    4540     0.621918  
19                        41875   15031     0.559939  
30                        17707    6267     0.547815  
2                          5841    1881     0.475000  
13                         5611    1545     0.379980  
5                         49174   13282     0.370055  
14                         3934    1030     0.354683  
23                         6100    1581     0.349856  
8                         29887    7012     0.306536  
11                         5081    1182     0.303155  
22                        13610    3150     0.301147  
25                         1978     450     0.294503  
24                         6777    1452     0.272676  
1                          2604     461     0.215119  
32                        11257    1927     0.206538  
15                         7950    1330     0.200906  
28                         6100    1016     0.199843  
29                         6506    1081     0.199263  
18                         3492     565     0.193030  
21                         4830     736     

In [8]:
# get population data
pop = pd.read_csv('./data/citydata/Community_Reporting_Areas_with_PL_94-171_Redistricting__Data_for_1990-2020.csv')
sum(pop['F2020_PL_data_TOT_POP'])

737015

In [9]:
print('UCV population:',sum(ucvpop['F2020_PL_data_TOT_POP']))
print('Total population:',sum(pop['F2020_PL_data_TOT_POP']))
print('UCV population fraction', round(sum(ucvpop['F2020_PL_data_TOT_POP'])/sum(pop['F2020_PL_data_TOT_POP']),3))

UCV population: 308307
Total population: 737015
UCV population fraction 0.418


In [10]:
print('UCV area (acres):',sum(ucvpop['AREA_ACRES']))
print('Total area (acres):',sum(pop['AREA_ACRES']))
print('UCV area fraction', round(sum(ucvpop['AREA_ACRES'])/sum(pop['AREA_ACRES']),3))

UCV area (acres): 16016.586737009999
Total area (acres): 53560.99999999999
UCV area fraction 0.299


In 2020, approximately 41.8% of the city population lived in an urban center or village while the urban villages only accounted for 29.9% of the city acreage.

## Kids

In [11]:
print('\nKids in UCVs\n')
print('2010 ',sum(ucvpop['F2010_PL_data_POP_UNDER_18']))
print('2020 ',sum(ucvpop['F2020_PL_data_POP_UNDER_18']))


Kids in UCVs

2010  21895
2020  27108


In [12]:
print('\nKids in entire city\n')
print('2010 ', sum(pop['F2010_PL_data_POP_UNDER_18']))
print('2020 ', sum(pop['F2020_PL_data_POP_UNDER_18']))


Kids in entire city

2010  93513
2020  106841


In [13]:
print("Fractional change in CHILD population in UCVs",
      round((sum(ucvpop['F2020_PL_data_POP_UNDER_18']) - 
             sum(ucvpop['F2010_PL_data_POP_UNDER_18']))/sum(ucvpop['F2010_PL_data_POP_UNDER_18']),
            3))
print()
print("Fractional change in CHILD population in entire city",
      round((sum(pop['F2020_PL_data_POP_UNDER_18']) - 
             sum(pop['F2010_PL_data_POP_UNDER_18']))/sum(pop['F2010_PL_data_POP_UNDER_18'])
            ,3), '\n')


Fractional change in CHILD population in UCVs 0.238

Fractional change in CHILD population in entire city 0.143 



In [14]:
print("Fractional change in ADULT population in UCVs", 
      round((sum(ucvpop['F2020_PL_data_POP_OVER_18']) - 
             sum(ucvpop['F2010_PL_data_POP_OVER_18']))/sum(ucvpop['F2010_PL_data_POP_OVER_18'])
            ,3))
print()
print("Fractional change in ADULT population in entire city",
      round((sum(pop['F2020_PL_data_POP_OVER_18']) - 
             sum(pop['F2010_PL_data_POP_OVER_18']))/sum(pop['F2010_PL_data_POP_OVER_18'])
            ,3),'\n')

Fractional change in ADULT population in UCVs 0.412

Fractional change in ADULT population in entire city 0.223 



From 2010 to 2020 the city as a whole saw an increase of 14.3% in the population of children.  Urban centers and villages saw a 23.8% rise in the child population over the same time period

Both UCVs and the city as a whole saw a larger fractional change in the adult population than in the child population.  The UCV adult population grew by 41.2% while the total adult population grew by 22.3% from 2010 to 2020.

In [15]:
ucvkids = ucvpop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in ucvpop.columns if 'UNDER' in col]]
kids = pop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in pop.columns if 'UNDER' in col]]

ucvadults = ucvpop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in ucvpop.columns if 'OVER' in col]]
adults = pop[['NEIGH_NAME','NEIGH_NO', 'AREA_ACRES']+[col for col in pop.columns if 'OVER' in col]]

In [16]:
ucvkids.columns = ['neighborhood','neighborhood_no','area','1990','2000','2010','2020']

In [17]:
ucvkids = ucvkids.copy()

In [18]:
ucvkids[ucvkids['neighborhood']=='Uptown']

neighborhood  neighborhood_no        area  1990  2000  2010  2020
OBJECTID                                                                  
20             Uptown              412  333.000698   172   147   222   596

In [19]:

ucvkids['change'] = ucvkids['2020']-ucvkids['2010']

In [20]:
ucvkids['frac_change'] = (ucvkids['2020']-ucvkids['2010'])/ucvkids['2010']

In [21]:
ucvkids['density'] = ucvkids['2020']/ucvkids['area']

In [22]:
ucvkids.sort_values('frac_change', ascending=False)

neighborhood  neighborhood_no         area  1990  2000  \
OBJECTID                                                                        
7                  South Lake Union              312   339.886927    80   122   
17        Ballard-Interbay-Northend              501   931.972864   107    66   
20                           Uptown              412   333.000698   172   147   
19                         Downtown              413   953.281747   276   755   
8               University District              401   751.963878   592   559   
12                 Greater Duwamish              502  4953.281793   148   213   
28                         Eastlake              107   199.255956   142   116   
14                       Green Lake              207   107.444638   103    94   
2                           Fremont              302   214.511083   266   171   
23              Bitter Lake Village              114   363.779566   379   396   
3             West Seattle Junction              205   270.417026   413   434   
4                     Columbia City              111   334.543340  1335  1423   
1                  Upper Queen Anne              106    52.658720   125   126   
6                         Roosevelt              202   170.211776   295   281   
30                          Ballard              303   495.476305   951   892   
9                        Crown Hill              103   270.988754   579   644   
11                        Lake City              201   142.267043   253   394   
29                      Wallingford              203   258.005613   448   477   
21                    Rainier Beach              204   346.143874  1159  1193   
27                  Morgan Junction              206   113.309377   191   243   
18          Greenwood-Phinney Ridge              102    94.157645   260   355   
25                          Admiral              113    98.248351   145   159   
13                   Madison-Miller              131   145.323420   480   334   
16                North Beacon Hill              110   266.876291  1001   992   
5           First Hill/Capitol Hill              414   916.221801  1729  1881   
22             23rd & Union-Jackson              109   625.266413  2036  2103   
15                        Northgate              301   412.291123   439   685   
26            Aurora-Licton Springs              101   326.958599   745   736   
24                         Mt Baker              305   491.018058   931  1016   
32                          Othello              123   498.963241  2338  2088   
10           Westwood-Highland Park              127   275.062196   970  1091   
31                       South Park              128   263.798620   638   855   

          2010  2020  change  frac_change   density  
OBJECTID                                             
7          158   549     391     2.474684  1.615243  
17          67   205     138     2.059701  0.219963  
20         222   596     374     1.684685  1.789786  
19         733  1607     874     1.192360  1.685756  
8          452   936     484     1.070796  1.244741  
12         135   245     110     0.814815  0.049462  
28         202   340     138     0.683168  1.706348  
14         163   265     102     0.625767  2.466386  
2          215   339     124     0.576744  1.580338  
23         597   929     332     0.556114  2.553744  
3          438   677     239     0.545662  2.503541  
4          968  1478     510     0.526860  4.417963  
1          136   201      65     0.477941  3.817032  
6          333   465     132     0.396396  2.731891  
30         964  1343     379     0.393154  2.710523  
9          747   927     180     0.240964  3.420806  
11         507   622     115     0.226824  4.372060  
29         701   839     138     0.196862  3.251867  
21        1087  1280     193     0.177553  3.697884  
27         268   294      26     0.097015  2.594666  
18         374   410      36     0.096257  4.354399  
25         198   215      

In [23]:
ucvkids.sort_values('2020', ascending=False).head(10)

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                     
32                        Othello              123  498.963241  2338  2088   
22           23rd & Union-Jackson              109  625.266413  2036  2103   
5         First Hill/Capitol Hill              414  916.221801  1729  1881   
19                       Downtown              413  953.281747   276   755   
4                   Columbia City              111  334.543340  1335  1423   
30                        Ballard              303  495.476305   951   892   
21                  Rainier Beach              204  346.143874  1159  1193   
24                       Mt Baker              305  491.018058   931  1016   
10         Westwood-Highland Park              127  275.062196   970  1091   
8             University District              401  751.963878   592   559   

          2010  2020  change  frac_change   density  
OBJECTID                                             
32        2759  2817      58     0.021022  5.645706  
22        1813  1902      89     0.049090  3.041903  
5         1786  1894     108     0.060470  2.067185  
19         733  1607     874     1.192360  1.685756  
4          968  1478     510     0.526860  4.417963  
30         964  1343     379     0.393154  2.710523  
21        1087  1280     193     0.177553  3.697884  
24        1046  1072      26     0.024857  2.183219  
10        1093   947    -146    -0.133577  3.442858  
8          452   936     484     1.070796  1.244741

In [24]:
ucvkids.sort_values('frac_change', ascending=False).head()

neighborhood  neighborhood_no        area  1990  2000  \
OBJECTID                                                                       
7                  South Lake Union              312  339.886927    80   122   
17        Ballard-Interbay-Northend              501  931.972864   107    66   
20                           Uptown              412  333.000698   172   147   
19                         Downtown              413  953.281747   276   755   
8               University District              401  751.963878   592   559   

          2010  2020  change  frac_change   density  
OBJECTID                                             
7          158   549     391     2.474684  1.615243  
17          67   205     138     2.059701  0.219963  
20         222   596     374     1.684685  1.789786  
19         733  1607     874     1.192360  1.685756  
8          452   936     484     1.070796  1.244741

In [25]:
ucvkids.sort_values('density', ascending = False)

neighborhood  neighborhood_no         area  1990  2000  \
OBJECTID                                                                        
32                          Othello              123   498.963241  2338  2088   
4                     Columbia City              111   334.543340  1335  1423   
11                        Lake City              201   142.267043   253   394   
18          Greenwood-Phinney Ridge              102    94.157645   260   355   
1                  Upper Queen Anne              106    52.658720   125   126   
21                    Rainier Beach              204   346.143874  1159  1193   
10           Westwood-Highland Park              127   275.062196   970  1091   
9                        Crown Hill              103   270.988754   579   644   
13                   Madison-Miller              131   145.323420   480   334   
29                      Wallingford              203   258.005613   448   477   
16                North Beacon Hill              110   266.876291  1001   992   
22             23rd & Union-Jackson              109   625.266413  2036  2103   
6                         Roosevelt              202   170.211776   295   281   
30                          Ballard              303   495.476305   951   892   
31                       South Park              128   263.798620   638   855   
27                  Morgan Junction              206   113.309377   191   243   
23              Bitter Lake Village              114   363.779566   379   396   
26            Aurora-Licton Springs              101   326.958599   745   736   
3             West Seattle Junction              205   270.417026   413   434   
14                       Green Lake              207   107.444638   103    94   
25                          Admiral              113    98.248351   145   159   
24                         Mt Baker              305   491.018058   931  1016   
15                        Northgate              301   412.291123   439   685   
5           First Hill/Capitol Hill              414   916.221801  1729  1881   
20                           Uptown              412   333.000698   172   147   
28                         Eastlake              107   199.255956   142   116   
19                         Downtown              413   953.281747   276   755   
7                  South Lake Union              312   339.886927    80   122   
2                           Fremont              302   214.511083   266   171   
8               University District              401   751.963878   592   559   
17        Ballard-Interbay-Northend              501   931.972864   107    66   
12                 Greater Duwamish              502  4953.281793   148   213   

          2010  2020  change  frac_change   density  
OBJECTID                                             
32        2759  2817      58     0.021022  5.645706  
4          968  1478     510     0.526860  4.417963  
11         507   622     115     0.226824  4.372060  
18         374   410      36     0.096257  4.354399  
1          136   201      65     0.477941  3.817032  
21        1087  1280     193     0.177553  3.697884  
10        1093   947    -146    -0.133577  3.442858  
9          747   927     180     0.240964  3.420806  
13         450   482      32     0.071111  3.316740  
29         701   839     138     0.196862  3.251867  
16         803   857      54     0.067248  3.211226  
22        1813  1902      89     0.049090  3.041903  
6          333   465     132     0.396396  2.731891  
30         964  1343     379     0.393154  2.710523  
31         855   685    -170    -0.198830  2.596678  
27         268   294      26     0.097015  2.594666  
23         597   929     332     0.556114  2.553744  
26         799   828      29     0.036295  2.532431  
3          438   677     239     0.545662  2.503541  
14         163   265     102     0.625767  2.466386  
25         198   215      17     0.085859  2.188332  
24        1046  1072      

Ballard-Interbay-Northend and Greater Duwamish are density outliers.  Both of these districts are huge industrial areas of the city that are covered in the urban village development plan.  

Ballard-Interbay-Northend includes the former WA National Guard site recently given over to the city.  Redeveloment options for this area are currently being explored by a legislatively appointed [advisory council](https://www.commerce.wa.gov/about-us/research-services/interbay-public-development-advisory-committee/).  

Greater Duwamish is the area including the port of Seattle.  

For the purpose of current data we can exclude them from the residential population analysis as they are mainly industrial rather than mixed-use, residential/commerical areas.  



In [26]:
#dropping Ballard-Interbay-Northend and Greater Duwamish areas
ucvkids.drop(index = 12, inplace = True)
ucvkids.drop(index = 17, inplace = True)

In [27]:
# saving urban center kid data
ucvkids.to_csv('./data/seattle_ucv_kids_cleaned.csv')

Inconsistency in definition of Urban villages and urban centers leads to oddities in the density values.  Uptown, Downtown, South Lake Union and First Hill/Capitol Hill create a contiguous block.  That is, the entirety of central Seattle is in one of these UCs, even though there are vast variations in density inside each of the UCs.

The definition of UC boundaries for those outside of central Seattle are much tighter, owing to strict zoning laws and the fight against expansion of non-single family housing.  Development in these areas has been limited to directly along the main thoroughfare with the urban center/village defined as such.  This leads to misleadingly high densities for these neighborhoods.  

Furthermore if we look at each of the Urban centers outside of central Seattle, they are all in close proximity to at least one if not several elementary schools.  

Kids experiencing homeless are considered residents of the Downtown UC and are shipped to Cap Hill for school.  This is in part due to the concentration of shelters in the downtown area and in part due to district policy.  



# To Do

- work on getting enrollment data in the correct form
- multiple dash figures on server
- dash: production server instead of development server?
- get data on moving rates?  get data on birth rates?
- get data on kid ages?

